src/ApiService/ApiService/Log.cs

```diff
@@ -1,418 +1,435 @@
-﻿using System.Diagnostics;
-using System.Net;
-using System.Runtime.CompilerServices;
-using System.Text;
+﻿using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Globalization;
 using Microsoft.ApplicationInsights;
 using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.Extensions.Logging;
 
 namespace Microsoft.OneFuzz.Service;
 
-//See: https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/tutorials/interpolated-string-handler
-//by ref struct works, but Moq does not support by-ref struct value so all the tests break... https://github.com/moq/moq4/issues/829
-[InterpolatedStringHandler]
-public struct LogStringHandler {
-
-    private readonly StringBuilder _builder;
-    private Dictionary<string, string>? _tags;
-
-    public LogStringHandler(int literalLength, int formattedCount) {
-        _builder = new StringBuilder(literalLength);
-        _tags = null;
-    }
-
-    public void AppendLiteral(string message) {
-        _builder.Append(message);
-    }
-
-    public void AppendFormatted<T>(T message) {
-        if (message is not null) {
-            _builder.Append(message.ToString());
-        } else {
-            _builder.Append("<null>");
-        }
-    }
-
-    public void AppendFormatted<T>(T message, string? format) {
-        if (format is not null && format.StartsWith("Tag:")) {
-            var tag = format["Tag:".Length..];
-            if (_tags is null) {
-                _tags = new Dictionary<string, string>();
-            }
-            _tags[tag] = $"{message}";
-            _builder.Append('{').Append(tag).Append('}');
-        } else if (message is IFormattable msg) {
-            _builder.Append(msg?.ToString(format, null));
-        } else {
-            _builder.Append(message?.ToString()).Append(':').Append(format);
-        }
-    }
-
-    private bool HasData => _builder is not null && _builder.Length > 0;
-
-    public override string ToString() => this.HasData ? _builder.ToString() : "<null>";
-    public IReadOnlyDictionary<string, string>? Tags => _tags;
+/// <summary>
```

Comment:
if there are no decriptions can we remove the <summary> comments.

Reply:
those are auto-populated/auto-created by Visual Studio

Reply:
Across all files, please 🙂